src/_pytest/capture.py

```diff
@@ -550,7 +550,7 @@ def snap(self) -> bytes:
         res = self.tmpfile.buffer.read()
         self.tmpfile.seek(0)
         self.tmpfile.truncate()
-        return res
+        return res  # type: ignore[return-value]
```

Comment:
It seems it's a false positive, the return from read() is in fact ``bytes`` and not a buffer ?

Reply:
Looks like it can be a `ReadableBuffer` and not `bytes`: https://github.com/python/typeshed/blob/ca65e08/stdlib/_io.pyi#L126C43-L126C57

Reply:
Should we fix something then (cast to str) ?

Reply:
The pre-commit bot removed our changes, I force pushed them back :)